content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md

```diff
@@ -0,0 +1,313 @@
+---
+title: "Dynamic Observation and Interruption of SPARQL Queries"
+date: 2024-03-29T03:19:19+01:00
+author: "Robin Textor-Falconi"
+authorAvatar: "img/ada.jpg"
+tags: ["SPARQL", "WebSocket", "HTTP", "Synchronization", "Atomics", "QLever"]
+categories: ["project"]
+image: "img/project-dynamic-observation-and-interruption-of-sparql-queries/teaser.png"
+draft: true
+---
+
+A dive into the process of designing an architecture that allows
+observers to interact with complex queries in real-time.
+
+<!--more-->
+
+## Content
+
+1. [Real-Time is more than quality of life](#real-time-is-more-than-quality-of-life)
+2. [Live query analysis for QLever](#live-query-analysis-for-qlever)
+    1. [Bidirectional Web Communication](#bidirectional-web-communication)
+    2. [Designing a WebSocket API](#designing-a-websocket-api)
+        1. [The runtime information tree](#the-runtime-information-tree)
+        2. [`Boost.Beast`, `Boost.Asio` and Concurrent connections](#boostbeast-boostasio-and-concurrent-connections)
+3. [Embracing Cancel Culture](#embracing-cancel-culture)
+    1. [Cancellation via WebSockets](#cancellation-via-websockets)
+    2. [Cancellation Handles and Watchdogs](#cancellation-handles-and-watchdogs)
+    3. [Collateral Benefits](#collateral-benefits)
+4. [Conclusion](#conclusion)
+
+
+## Real-Time is more than quality of life
+
+QLever is a SPARQL-compliant knowledge-base engine that allows to blazingly fast execution of SPARQL queries on giant RDF
```

Comment:
typo "tthat allows *the* blazingly fast execution" or "that allows to blazingly fast execute"